### PR TITLE
[SYCL] Use full compilation for the multi-device AOT test

### PR DESCRIPTION
**Pull request by AGindinson**
Since the introduction of the original test, many changes have been
made:
- the tests' runtime and the compilation time have been reasonably
  improved;
- the compiler pipeline has been complicated by a bunch of DPC++
  tools;
- default compiler behavior for many features includes dependency on
  device libraries/injections by new DPC++ tools, which makes the test
  increasingly hard to maintain.

Therefore, it makes sense to refrain from running the compiler steps
manually from now on, relying on E2E compilation as our users
would. We will produce a single fat object for all supported device
targets + host, and then rely on the driver/tools to unbundle the
object & compile only for the needed targets.

Signed-off-by: Artem Gindinson <artem.gindinson@intel.com>

### CODE REVIEW

File: SYCL/AOT/multiple-devices.cpp [hunk 1]
```diff
@@ -10,70 +10,50 @@
 // UNSUPPORTED: cuda
```
Comment (AlexeySachkov) [inline SYCL/AOT/multiple-devices.cpp:10]: I assume we can update `REQUIRES` now to exclude a couple of tools from there, right?

**Review by AlexeySachkov — DISMISSED** [review]

File: SYCL/AOT/multiple-devices.cpp (hunk 1 above)
Comment (AGindinson) [reply SYCL/AOT/multiple-devices.cpp:10]: Good idea, done in https://github.com/intel/llvm-test-suite/pull/603/commits/dd9c661e468f7c612d96297cc44a46acef5fa680

**Comment by AGindinson** [comment]
Windows-specific runtime failures in `DeviceLib` tests don't seem related.

**Review by AlexeySachkov — APPROVED** [review]

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
llvm-spirv is also used in https://github.com/intel/llvm-test-suite/blob/intel/SYCL/AOT/with-llvm-bc.cpp. Could you please update the test accordingly?

**Comment by AGindinson** [comment]
> llvm-spirv is also used in https://github.com/intel/llvm-test-suite/blob/intel/SYCL/AOT/with-llvm-bc.cpp. Could you please update the test accordingly?

llvm-spirv has not been marked as `REQUIRED` there (neither has it been in `Config/kernel-from-file.cpp`). Also, `AOT/with-llvm-bc` can't really be updated to use driver calls only, since its point is not to have SPIR-V in the final binary at all. Are the changes proposed in this PR really blocked by that separate test?

Or is the point to `REQUIRE` the translator in these test instead? To be frank, I don't really understand the point of checking the tools' availability, since our compiler wouldn't function for the rest of the tests without them.

**Review by vladimirlaz — APPROVED** [review]

**Review by mdtoguchi — APPROVED** [review]
LGTM

**Pull request merged**